the-collab-lab/tcl-8-smart-shopping-list
# Lb sa estimated purchase time

## Description

This PR divides the list into active and inactive items, then sorts the list according to the ACs, adds a new property of 'status', and depending on the status, adds styling.
NPM package [day.js](https://day.js.org/) was utilized to calculate and format dates accordingly. 

## Acceptance Criteria

- Items in the list are shown visually grouped in the following order: Soon, Kind of soon, Not soon, Inactive
- Within each grouping, items should be sorted by the estimated number of days until next purchase
- Items with the same number of estimated days until next purchase should be sorted alphabetically
- Items in the different states should be visually distinct
- Items in the different states should be described distinctly when read by a screen reader

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|✓| :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|✓| :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### After

<img width="1365" alt="Screen Shot 2020-07-11 at 9 53 05 PM" src="https://user-images.githubusercontent.com/45564632/87237175-06e50c00-c3c1-11ea-94b6-df5a4884d39a.png">

## Testing Steps / QA Criteria

To test the functionality: 
1. You can use the following token that has all of the relevant item status occurrences: `lucky cicada paean`
2. You can change item properties in the database to see items are sorted accordingly. 
    a. If `numberOfPuchases` <= 1 -OR- if elapsed time since the `lastPurchasedDate` is 2x `nextPurchase` then Item is inactive and should appear at the bottom of the list.
    b. The rest of the items are sorted by the number of `numberOfPurchase` days and also alphabetically. 

## Todo

Working on improving styling to make the item status more distinct
